SourceryRuntime/Sources/Extensions.swift

```diff
@@ -130,47 +135,46 @@ public extension String {
 
     /// :nodoc:
     func components(separatedBy delimiter: String, excludingDelimiterBetween between: (open: String, close: String)) -> [String] {
+        return self.components(separatedBy: delimiter, excludingDelimiterBetween: (between.open.map { String($0) }, between.close.map { String($0) }))
+    }
+
+    /// :nodoc:
+    func components(separatedBy delimiter: String, excludingDelimiterBetween between: (open: [String], close: [String])) -> [String] {
         var boundingCharactersCount: Int = 0
         var quotesCount: Int = 0
         var item = ""
         var items = [String]()
         var matchedDelimiter = (alreadyMatched: "", leftToMatch: delimiter)
 
-        for char in self {
-            if between.open.contains(char) {
-                if !(boundingCharactersCount == 0 && String(char) == delimiter) {
+        var i = self.startIndex
+        while i < self.endIndex {
+            var increaseOffset = 1
+            defer {
+                i = self.index(i, offsetBy: increaseOffset)
+            }
+            let current = self[i..<(self.index(i, offsetBy: delimiter.count, limitedBy: self.endIndex) ?? self.endIndex)]
+            if let index = between.open.index(where: { String(self[i...]).starts(with: $0) }) {
```

Comment:
This seems to find the opening delimiter that string suffix starts with, but the index is then not being used for anything but getting this delimiter, so I guess it could be access here with `between.open.first(where:)` which will make the intention more clear.
Also `String(self[i...])` should be calculated only once.

Reply:
Makes sense to me. Will update